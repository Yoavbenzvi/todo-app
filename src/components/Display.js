import React from 'react';
import Task from './Task';

const Display = ({ tasks, showAll, toggleTaskStatus, deleteTask }) => {

	const renderText = () => {
		return tasks.length <= 0 ?
			<span className='text-center inline-block w-full p-4 text-gray-600 text-lg'>
				Add a new item to get started!
			</span>
			:
			null
	}

	const renderList = () => {
		let allTasks;

		if(showAll) {
			allTasks = tasks.map((task, i) => {
				return <Task 
					status={task.status}
					text={task.text}
					toggleTaskStatus={toggleTaskStatus}
					index={i}
					key={i}
					deleteTask={deleteTask}
				/>
			})
		} else {
			allTasks = tasks.filter(task => task.status).map((task, i) => {
				return <Task 
					status={task.status}
					text={task.text}
					toggleTaskStatus={toggleTaskStatus}
					index={i}
					key={i}
					deleteTask={deleteTask}
				/>
			})
		}

		return allTasks
	}

	return(
		<div className='h-64 overflow-y-scroll'>
			<ul
				className='m-0 my-4 p-0 list-none w-full'
			>
				{renderList()}
			</ul>
			{renderText()}
		</div>
	)
}

export default Display