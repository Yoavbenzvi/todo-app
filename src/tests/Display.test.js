import { shallow } from "enzyme";
import React from "react";
import Display from "../components/Display.js";

describe('Display component', () => {
	it("expect to render Display component", () => {

		const wrapper = shallow(<Display tasks={[]}/>);
		expect(wrapper).toMatchSnapshot();
	});
})
