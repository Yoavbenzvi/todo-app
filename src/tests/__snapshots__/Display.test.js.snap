// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component expect to render Display component 1`] = `
<div
  className="h-64 overflow-y-scroll"
>
  <ul
    className="m-0 my-4 p-0 list-none w-full"
  />
  <span
    className="text-center inline-block w-full p-4 text-gray-600 text-lg"
  >
    Add a new item to get started!
  </span>
</div>
`;
